### PR TITLE
Feature/monitored stack

### PR DESCRIPTION
**Pull request by vdektiarev**
1. Add eventbridge roles and policies
2. Add rules for eventbridge in monitored stacks - glue and step functions
3. Formatting and folder reorganization
4. General sample file structure and naming change + adaptation in readers and tests

### CODE REVIEW

**Review by NastassiaPaskrobka — CHANGES_REQUESTED** [review]

File: infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py [hunk 1]
```diff
@@ -9,86 +9,122 @@
...
+                    e.doc,
+                    e.pos,
+                )
+
+        monitored_account_ids = [
+            account["AccountId"] for account in general_config["monitored_environments"]
+        ]
+        monitored_principals = [
+            iam.AccountPrincipal(account_id) for account_id in monitored_account_ids
+        ]
```
[the remark this answers is not in the document]
Comment (natmasslove) [reply infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py:73]: I agree. For me it looks like it should be a call to settings component

File: infra_monitored_account/infra_monitored_account/infra_monitored_stack.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    aws_events_targets as targets,
+)
+from constructs import Construct
+import json
+
+
+class InfraMonitoredStack(Stack):
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        stage_name = kwargs.pop("stage_name", None)
+        project_name = kwargs.pop("project_name", None)
```
[the remark this answers is not in the document]
Comment (natmasslove) [reply infra_monitored_account/infra_monitored_account/infra_monitored_stack.py:15]: In this case, pop is needed (we should remove some keys before passing kwargs to super's construct)

File: config/sample_settings/general.json [hunk 3]
```diff
@@ -1,13 +1,20 @@
 {
-    "monitored_accounts": [
+    "tooling_environment": {
+        "Name": "Tooling Account",
+        "AccountId": "323432554",
+        "Region": "eu-central-1"
+    },
+    "monitored_environments": [
```
Comment (NastassiaPaskrobka) [inline config/sample_settings/general.json:7]: Missing changes to monitoring_groups.json with added "monitored_environment" property

File: config/sample_settings/general.json [hunk 4]
```diff
@@ -1,13 +1,20 @@
 {
-    "monitored_accounts": [
+    "tooling_environment": {
+        "Name": "Tooling Account",
```
Comment (natmasslove) [inline config/sample_settings/general.json:3]: Let's also name it like "Tooling Account [dev]" to demonstrate the config approach we suggest to future users

File: infra_monitored_account/app.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+from infra_monitored_account.infra_monitored_stack import InfraMonitoredStack
+
+if "STAGE_NAME" in os.environ:
+    pass
+else:
+    exec('raise ValueError("Environment variable STAGE_NAME is undefined")')
+stage_name = os.environ["STAGE_NAME"]
+
+project_name = "salmon"
```
[the remark this answers is not in the document]
Comment (vdektiarev) [reply infra_monitored_account/app.py]: Fixed

File: infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vdektiarev) [reply infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py]: Decided to brainstorm that in future pull requests

File: config/sample_settings/general.json (hunk 3 above)
Comment (vdektiarev) [reply config/sample_settings/general.json:7]: Fixed

File: config/sample_settings/general.json (hunk 4 above)
Comment (vdektiarev) [reply config/sample_settings/general.json]: Added

File: infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py [hunk 6]
```diff
@@ -9,86 +9,122 @@
...
+                general_config = json.load(f)
+            except json.decoder.JSONDecodeError as e:
+                raise json.decoder.JSONDecodeError(
+                    f"Error parsing JSON file {general_settings_file_path}",
+                    e.doc,
+                    e.pos,
+                )
+
+        monitored_account_ids = [
+            account["AccountId"] for account in general_config["monitored_environments"]
```
Comment (natmasslove) [inline infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py:69]: there might be a case when we have 2 monitored env (same account, different regions).
I guess, we might need remove duplicates here

File: infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py [hunk 7]
```diff
@@ -9,86 +9,122 @@
...
-            rule_name=f"eventbusrule-{
-                project_name}-notification-lambda-{stage_name}",
+            rule_name=f"eventbusrule-{project_name}-notification-lambda-{stage_name}",
             event_bus=notification_bus,
-            event_pattern=events.EventPattern(source=events.Match.prefix("aws"))
+            event_pattern=events.EventPattern(source=events.Match.prefix("aws")),
         )
 
         # Notification Lambda Role
         notification_lambda_role = iam.Role(
```
Comment (natmasslove) [inline infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py:95]: based on sync-up conversation - looks like we need to rename it to alerts-lambda as notification-lambda is a different entity which actually sends notifications

**Review by NastassiaPaskrobka — APPROVED** [review]

**Review by natmasslove — APPROVED** [review]

File: infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py (hunk 6 above)
Comment (vdektiarev) [reply infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py]: Fixed

**Pull request merged**